src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -83,6 +83,20 @@ public IReadOnlyNetworkPeerCollection ConnectedPeers
 
         private AsyncQueue<INetworkPeer> connectedPeersQueue;
 
+        /// <summary>Traffic statistics from peers that have been disconnected.</summary>
+        private PerformanceCounter counter;
+
+        public PerformanceCounter Counter
+        {
+            get
+            {
+                if (this.counter == null)
```

Comment:
Is thread safety important here

Reply:
Very small impact, but If you initialize it in the constructor then probably not.